### PR TITLE
Check for UNKNOWN task type

### LINKED ISSUE #215: Use CMS_extendedTaskType to support physics job task types
At present, physics task types are supported via:

https://github.com/dmwm/cms-htcondor-es/blob/master/src/htcondor_es/convert_to_json.py#L682-L684

WMCore has recently added support for this. The classad is called: `CMS_extendedJobType` and it shows the physics types when this is a production/processing job, or just the WMCOre job type (LogCollect, Merge, etc) when it is not.

We would need to replace the current `CMS_TaskType` in the code lines above with this new classad `CMS_extendedJobType` and still fallback to the guessTaskType function until this feature is fully propagated.
 Do I understand correctly that `CMS_TaskType` was a placeholder for a future WMCore native implementation of this by the way? I don't see this classad in the jobs in condor. 

Documentation can be found below:
https://github.com/dmwm/WMCore/wiki/Job-task-type-characterization-based-on-cmsDriver-command-line-arguments

But it basically support the following physics types (and a combination of them)

Physics types:

- GEN
- SIM
- DIGI_nopileup
- DIGI_premix
- DIGI_classicalmix
- RECO
- MINIAOD
- NANOAOD
- UNKNOWN

### PR DESCRIPTION
**Pull request by nikodemas**
Should close https://github.com/dmwm/cms-htcondor-es/issues/215.

Update `TaskType` calculation so that when `CMS_extendedJobType` is `UNKNOWN`, `TaskType` would be derived from the other fields (previously it only happened when `CMS_extendedJobType` was `None`).

FYI @leggerf @brij01

### CODE REVIEW

**Review by leggerf — APPROVED** [review]

**Comment by leggerf** [comment]
maybe also add in plain English an explanation about the implemented logic, so that we do not have to figure it out every time we look at the code ;)

**Comment by nikodemas** [comment]
Do you think that the comment is necessary? Because I think that I would simply be repeating the code with a comment like
`# if CMS_extendedJobType is UNKNOWN or None, TaskType is derived from the other fields`
and I cannot think of anything more meaningful :/

**Comment by leggerf** [comment]
well, I was thinking something among the lines: 'TaskType is filled with CMS_extendedJobType. If CMS_extendedJobType is not filled or UNKNOWN it is set to CMS_TaskType. If this is still empty, it is set to CMS_jobType for analysis jobs or to a combination of CMS_jobType and WMAgent_SubTaskName for production jobs (see guessTaskType function)'

obviously the ultimate reference is the code itself, but I think it helps have some sort of textual description

**Comment by nikodemas** [comment]
https://cmsmonit-docs.web.cern.ch/cms-htcondor-es/cms-htcondor-es/#name-tasktype could I maybe update it here in the documentation instead and leave the code without the comment?
